master/buildbot/changes/bitbucket.py

```diff
@@ -14,184 +14,282 @@
 # Copyright Buildbot Team Members
 
 
-import json
-import time
 from datetime import datetime
 
 from twisted.internet import defer
-from twisted.python import log
-from twisted.web import client
 
 from buildbot import config
 from buildbot.changes import base
 from buildbot.util import bytes2unicode
 from buildbot.util import datetime2epoch
-from buildbot.util import deferredLocked
-from buildbot.util import epoch2datetime
+from buildbot.util import httpclientservice
+from buildbot.util.logger import Logger
+from buildbot.util.state import StateMixin
 
-_UNSPECIFIED = object()
+log = Logger()
 
+BASE_URL = "https://api.bitbucket.org"
+OAUTH_URL = 'https://bitbucket.org/site/oauth2/access_token'
+_GET_TOKEN_DATA = {
+    'grant_type': 'client_credentials'
+}
 
-class BitbucketPullrequestPoller(base.PollingChangeSource):
 
-    compare_attrs = ("owner", "slug", "branch",
-                     "pollInterval", "useTimestamps",
-                     "category", "project", "pollAtLaunch")
+class BitbucketPullrequestPoller(base.ReconfigurablePollingChangeSource,
+                                 StateMixin):
 
-    db_class_name = 'BitbucketPullrequestPoller'
+    compare_attrs = ("owner", "slug", "branches", "pollInterval",
+                     "category", "pollAtLaunch", "name")
+    db_class_name = "BitBuckerPullrequestPoller"
 
-    def __init__(self, owner, slug,
-                 branch=None,
-                 pollInterval=10 * 60,
-                 useTimestamps=True,
-                 category=None,
-                 project='',
-                 pullrequest_filter=True,
-                 encoding=_UNSPECIFIED,
-                 pollAtLaunch=False
-                 ):
+    def __init__(self, owner, slug, **kwargs):
+        name = kwargs.get("name")
```

Comment:
two comments about these lines:
* could we use `kwargs.setdefault("name", ...)` instead?
* could you please replace string concatenations with a string interpolation (`"xxx {} xxx {} xxx`.format(...)`)? interpolations are easier to read and easier to review when changes are made